Terminal.Gui/Views/ComboBox.cs

```diff
@@ -101,7 +132,20 @@ public ComboBox (ustring text) : base ()
 		public ComboBox (Rect rect, IList source) : base (rect)
 		{
 			search = new TextField ("") { Width = rect.Width };
-			listview = new ListView (rect, source) { LayoutStyle = LayoutStyle.Computed, ColorScheme = Colors.Base };
+			listview = new ComboListView (rect, source, HideDropdownListOnClick) { LayoutStyle = LayoutStyle.Computed, ColorScheme = Colors.Base };
+
+			Initialize ();
+			SetSource (source);
+		}
+
+		/// <summary>
+		/// Initialize with the source.
```

Comment:
If a user chooses to NOT use this constructor, but uses the parameterless constructore and sets the `Source` property instead, will the ComboBox work the same?

Generally, parameterless constructors are preferred but this PR seems to be biased towards having people use constructors with params.

Reply:
> If a user chooses to NOT use this constructor, but uses the parameterless constructore and sets the `Source` property instead, will the ComboBox work the same?

Yes it will work the same without any issue because this new constructor isn't used.


> Generally, parameterless constructors are preferred but this PR seems to be biased towards having people use constructors with params.

This doesn't affect the normal execution of the view. I only created it to be consistent with `ListView(IList source)` constructor which the `ComboBox` already had but with the `Rect` parameter. So, it's obvious the this `Computed` constructor is great.

Reply:
> > If a user chooses to NOT use this constructor, but uses the parameterless constructore and sets the `Source` property instead, will the ComboBox work the same?
> 
> Yes it will work the same without any issue because this new constructor isn't used.

Sorry,  but my question was not as clear as I meant it:

If a user chooses to NOT use this constructor, but uses the parameterless constructor and sets the `Source` property instead, will the ComboBox work the SAME AS IF THE USER used the new constructor?

Reply:
No, it won't, because the `Source` property only accepts a `IListDataSource`. For `IList` type the `Source` property only can be set by the `SetSource` method. The new constructor is more to work directly to set the `ListView` source by using the same `IList` source parameter.

Reply:
> No, it won't, because the `Source` property only accepts a `IListDataSource`. For `IList` type the `Source` property only can be set by the `SetSource` method.

My bad. I forgot t mention it can be set by calling the `ListWrapper` constructor directly:
`listView.Source = new ListWrapper (null);`

I prefer this approach because we can have a null source but still having a non null `Source` instance.

Reply:
> If a user chooses to NOT use this constructor, but uses the parameterless constructor and sets the `Source` property instead, will the ComboBox work the SAME AS IF THE USER used the new constructor?

Sorry for my wrong response before. Yes it will if the `Source` is initialize using the `ListWrapper` constructor but with the only difference that if it's set with the `SetSource` method with a null value the `IListDataSource` instance will not be created and using the `ListWrapper` constructor with a null parameter a `IListDataSource` instance will be created with a null source.